Library/Homebrew/patch.rb

```diff
@@ -127,9 +127,22 @@ def owner=(owner)
   def apply
     dir = Pathname.pwd
     resource.unpack do
-      # Assumption: the only file in the staging directory is the patch
-      patchfile = Pathname.pwd.children.first
-      dir.cd { safe_system "/usr/bin/patch", "-g", "0", "-f", "-#{strip}", "-i", patchfile }
+      patch_dir = Pathname.pwd
+      children = patch_dir.children
+      if patch_files.empty?
+        if (children.count == 1 && children.first.file?)
+          patch_files << children.first.basename
+        else
+          raise("There should be exactly one patch file in the staging directory unless the \"apply\" method was used one or more times in the patch-do block.")
+        end
+      end
+      dir.cd do
+        patch_files.each do |patch_file|
+          ohai "Applying #{patch_file}"
```

Comment:
Could you give an example, on what will it look like on normal mode and verbose mode?